### PR TITLE
Feature/remote tags

### LINKED ISSUE #216: Support for some remote tags, and some local-only tags
Users may wish to specify certain image tags as "remote" tags, and other image tags as "local" tags. If "remote" tags are provided, the `pull` and `push` tasks should execute _only_ on the "remote" tags. All other tasks should execute on both "remote" and "local" tags.

### PR DESCRIPTION
**Pull request by aidan-mundy**
Closes https://github.com/dnephin/dobi/issues/216

### CODE REVIEW

File: tasks/image/image.go
Comment (dnephin) [inline tasks/image/image.go:75]: What does `NoFallback` in the name mean here? 

I often try to avoid functions that only call one other function, especially when there are other functions with similar names. I guess each of the callers of this function could instead directly call:

```go
t.forEachProvidedTag(each, t.config.RemoteTags)
```

**Review by dnephin — APPROVED** [review]
Thanks! LGTM

I think this is a good way of handling backwards compat. I left a very minor style suggestion, but I'll merge this as-is, and leave it to you if you think it's worth changing in a follow up.

File: tasks/image/image.go
Comment (aidan-mundy) [reply tasks/image/image.go:75]: "NoFallback" in this case refers to the lack of "fallback" to Local tags when there are no remote tags. I generally agree on avoiding "functions that only call one other function", except when the call is (and always should be) exactly the same across multiple callers. 

Initially, I did not have this function at all, and `forEachTag` instead called `forEachRemoteTag`, causing local tags to be applied twice if there were no remote tags. Using a call to `forEachProvidedTag` entices a later (and more forgetful) me to go back and "DRY" the code by changing it back to `forEachRemoteTag` 🤦‍♂️. I figure that this is a case of trading actually DRY code for slightly more nested depth in the codebase. Perhaps `forEachRemoteTagNoFallbackToLocal` would be a more clear name though?

**Pull request merged**
